backend/src/apiserver/main.go

```diff
@@ -72,12 +83,59 @@ func main() {
 		template.Launcher = common.GetStringConfig(launcherEnv)
 	}
 
-	clientManager := cm.NewClientManager()
+	logLevel := *logLevelFlag
+	if logLevel == "" {
+		logLevel = "info"
+	}
+
+	level, err := log.ParseLevel(logLevel)
+	if err != nil {
+		log.Fatal("Invalid log level:", err)
+	}
+	log.SetLevel(level)
+
+	zapLevel, err := common.ParseLogLevel(logLevel)
+	if err != nil {
+		glog.Infof("%v. Defaulting to info level.", err)
+		zapLevel = zapcore.InfoLevel
+	}
+
+	ctrllog.SetLogger(zap.New(zap.UseFlagOptions(&zap.Options{Level: zapLevel})))
+
+	clientManager, err := cm.NewClientManager()
+	if err != nil {
+		glog.Fatalf("Failed to initialize ClientManager: %v", err)
+	}
+
+	defer clientManager.Close()
+
+	backgroundCtx, backgroundCancel := context.WithCancel(signals.SetupSignalHandler())
+	defer backgroundCancel()
+
+	wg := sync.WaitGroup{}
+
+	wg.Add(1)
+	webhookServer, err := startWebhook(clientManager.ControllerClient(), &wg)
```

Comment:
right now it looks like we are always starting the server, which seems unnecessary when k8s native mode is not enabled, so it makes more sense to me to introduce this flag in this pr first so we can make this conditional, if we are merging this first